File: sample/cukescala/src/test/scala/scaladefs/CalculatorStepDefs.scala
package scaladefs

import cucumber.api.scala.{EN, ScalaDsl}
import cucumber.api.Scenario
import calculator.RpnCalculator

class CalculatorStepDefs extends ScalaDsl with EN {
  val calc = new RpnCalculator

  When( """^I add (\d+) and (\d+)$""") {
    (arg1: Double, arg2: Double) =>
      calc push arg1
      calc push arg2
      calc push "+"
  }

  Then("^the result is (\\d+)$") {
    expected: Double =>
      assertEquals(expected, calc.value, 0.001)
  }
}

File: sample/cukejava/src/test/java/javadefs/CalculatorStepdefs.java
package javadefs;

import cucumber.api.java.en.Then;
import cucumber.api.java.en.When;

import static org.junit.Assert.assertEquals;

public class CalculatorStepdefs {

    private int result = 0;

    @When("^I add (\\d+) and (\\d+)$")
    public void addTwoNumbers(int arg1, int arg2) throws Throwable {
        result = arg1 + arg2;
    }

    @Then("^the result is (\\d+)$")
    public void assertResult(int expected) throws Throwable {
        assertEquals(expected, result);
    }
}
